pennylane/transforms/hamiltonian_expand.py

```diff
@@ -106,26 +106,34 @@ def hamiltonian_expand(tape, group=True):
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
 
-    if group:
-        hamiltonian.simplify()
-        return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
+    if group or hamiltonian.grouping_indices is not None:
+        # if explicitly requested or already available,
+        # compute the grouping information
+        coeffs_groupings, obs_groupings = hamiltonian.get_groupings()
+    else:
+        # else make each observable its own group
+        obs_groupings = [[obs] for obs in hamiltonian.ops]
+        coeffs_groupings = [hamiltonian.coeffs[i] for i in range(len(hamiltonian.ops))]
 
     # create tapes that measure the Pauli-words in the Hamiltonian
     tapes = []
-    for ob in hamiltonian.ops:
-        # we need to create a new tape here, because
-        # updating metadata of a copied tape is error-prone
-        # when the observables were changed
+    for obs in obs_groupings:
+
         with tape.__class__() as new_tape:
             for op in tape.operations:
-                qml.apply(op)
-            qml.expval(ob)
+                op.queue()
+
+            for o in obs:
+                qml.expval(o)
+
+        new_tape = new_tape.expand(stop_at=lambda obj: True)
```

Comment:
I have no clue why, but when I take this line out a few tests fail. I tried to compare the expanded tape with the non-expanded one, and the latter has no queue, but else they look the same. Strange!

The line is from @Lucaman99 I think?

Reply:
Ah, that may be it, will try!

Reply:
I did remove the line without issues - maybe it was different bugs interfering previously!

Reply:
I assume it is this computation of diagonalisations that is crucial in `tape_expand`:

```
    if tape._obs_sharing_wires:
        try:
            rotations, diag_obs = qml.grouping.diagonalize_qwc_pauli_words(tape._obs_sharing_wires)
        except ValueError as e:
            raise qml.QuantumFunctionError(
                "Only observables that are qubit-wise commuting "
                "Pauli words can be returned on the same wire"
            ) from e

        tape._ops.extend(rotations)

        for o, i in zip(diag_obs, tape._obs_sharing_wires_id):
            new_m = qml.measure.MeasurementProcess(tape.measurements[i].return_type, obs=o)
            tape._measurements[i] = new_m
```

Reply:
ohhhh